src/infoManagers/endpointManager.ts

```diff
@@ -37,7 +37,12 @@ export class EndpointManager extends Disposable {
 	 * @param location the file location to encode.
 	 * @returns the encoded endpoint.
 	 */
-	public async encodeLooseFileEndpoint(location: string): Promise<string> {
+	public async encodeLooseFileEndpoint(file: string | vscode.Uri): Promise<string> {
+		if (typeof file !== 'string' && file.scheme === 'vscode-chat-code-block') {

```

Comment:
In theory we could try to handle all types of URIs that are not loose files on disk or untitled files, instead of just handling `vscode-chat-code-block` here but I didn't have enough confidence in what this method has to handle